src/rgw/motr/sync/motr_sync_impl.cc

```diff
@@ -0,0 +1,149 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * GC thread/process synchronization for the CORTX Motr backend
+ *
+ * Copyright (C) 2022 Seagate Technology LLC and/or its Affiliates
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include "motr/sync/motr_sync_impl.h"
+
+void MotrLock::initialize(std::shared_ptr<MotrLockProvider> lock_provider) {
+  _lock_provider = lock_provider;
+}
+
+int MotrLock::lock(const std::string& lock_name, MotrLockType lock_type,
+                   utime_t lock_duration, const std::string& locker_id) {
+  if (!_lock_provider ||
+      ((MotrLockType::EXCLUSIVE == lock_type) && locker_id.empty())) {
+    return -EINVAL;
+  }
+  int rc = 0;
+  motr_lock_info_t lock_obj;
+  rc = _lock_provider->read_lock(lock_name, &lock_obj);
+  if (rc == -ENOENT) {
+    // lock is available/free
+  } else {
+    if (MotrLockType::EXCLUSIVE == lock_type) {
+      // caller needs exclusive lock...but lock entry is present
+      return -EBUSY;
```

Comment:
Before returning EBUSY, first check if the lock has been expired. If not expired, then return EBUSY, else delete the lock & try to acquire again.